plugins/inputs/execd/shim/goshim_notwindows.go

```diff
@@ -0,0 +1,14 @@
+// +build !windows
+
+package shim
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+func listenForCollectMetricsSignals(collectMetricsPrompt chan os.Signal) {
+	// just listen to all the signals.
+	signal.Notify(collectMetricsPrompt, syscall.SIGHUP, syscall.SIGUSR1, syscall.SIGUSR2)
```

Comment:
It's a convention for daemon processes to reload config on HUP.  Even if the shim doesn't do that yet I think we ought to avoid using HUP as a collection signal in case we want to use it to reload config in the future.

Reply:
yeah I don't like it either. Unfortunately I inherited this from whoever did the first execd version where they decided they should let the user pick how to signal metric handling, providing SIGHUP as an option. So if we remove it we have to agree we're okay with breaking the last published version of execd input.